### PR TITLE
141 add search display option

### PR DESCRIPTION
**Pull request by michael-conway**
Merge latest into master, this should be super close

### CODE REVIEW

**Comment by trel** [comment]
can we reconcile the two metalnx.properties files?   or even better, remove one of them?

**Comment by michael-conway** [comment]
Yes I'd get rid of the top level /etc/irods-ext and instead point folks to
the /etc/irods-ext under the docker-test-framework. That was annoying me
too.

NB that docker-test-framework is mirrored in jargon and in the metadata
search plugin projects so if you bring up irods in jargon, metadata search
in that framework, and metalnx all the components will find and talk to
each other with the default config.

**Comment by trel** [comment]
okay, so please remove the /etc/irods-ext and use/mention the files in the testing framework instead.

Please restore the multistage Dockerfile, and then squash all this work into 2-3 useful commits.

Instead of having two booleans for the different searches (where both could be visible...), should we just put search into a 'mode'?   Define search as either in 'classic/avu/genquery' or 'pluggable'?   Eventually, we'd ship a plugin to replace the current 'classic/avu/genquery' search?   So all search would be via plugins?  And then the default setting would change to be just 'search=pluggable'?

**Comment by michael-conway** [comment]
Hey Terrell there can be cases where you have a combo of the classic and the new pluggable, it's really only when we get a sort of pre-baked indexing and avu search that we can get rid of the 'classic', otherwise they can be present in any combo -> the dual boolean seems the cleanest...

Is that multi-stage Docker of any utility? It's not used in compose at all. I thought this was a cause of confusion for @deeppatelcs ?

**Comment by trel** [comment]
The multistage Dockerfile was a community contribution - and it could replace the main one once we vet it a bit more and confirm it's behaving as expected.

**Comment by trel** [comment]
And okay on the double boolean.  Thanks.

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ psql> GRANT ALL PRIVILEGES ON DATABASE "IRODS-EXT" TO metalnx;
  - Configuration of Zone information, and features to display
  - Theming with custom CSS/Logo
 
-Create a copy of the default `etc/irods-ext` directory and update `metalnx.properties` and `metalnxConfig.xml`, and then run a container with the new configuration, probably with `--add-host` information due to Docker:
+Create a copy of the default `/docker-test-framework/etc/irods-ext` directory and update `metalnx.properties` and `metalnxConfig.xml`, and then run a container with the new configuration, probably with `--add-host` information due to Docker:
```
Comment (trel) [inline README.md:34]: i think remove the leading slash

**Comment by trel** [comment]
can we sniff and more elegantly complain about missing configuration options if they're required?   if they're not required, can we have metalnx not blow up by using a strong default and just putting a notice in the log that it was not defined in the config?

**Comment by michael-conway** [comment]
Yes I think this is a feature, like a properties migration utility.

**Comment by trel** [comment]
I mean, if we're introducing a new property - provide a default and logging so the app can stand up with old configs.

**Pull request merged**